web/src/lib/components/shared-components/search-bar/search-location-section.svelte

```diff
@@ -1,12 +1,62 @@
+<script lang="ts" context="module">
+  export interface SearchLocationFilter {
+    country?: string;
+    state?: string;
+    city?: string;
+  }
+</script>
+
 <script lang="ts">
+  import { getSearchSuggestions, SearchSuggestionType } from '@immich/sdk';
   import Combobox, { type ComboBoxOption } from '../combobox.svelte';
   import type { SearchFilter } from './search-filter-box.svelte';
 
-  export let filteredLocation: SearchFilter['location'];
-  export let suggestedCountries: ComboBoxOption[];
-  export let suggestedStates: ComboBoxOption[];
-  export let suggestedCities: ComboBoxOption[];
-  export let updateSuggestions: () => Promise<void>;
+  export let filter: SearchFilter['location'];
+
+  let countries: string[] = [];
+  let states: string[] = [];
+  let cities: string[] = [];
+
+  const toComboBoxOptions = (items: string[]) => items.map<ComboBoxOption>((item) => ({ label: item, value: item }));
+
+  $: countryFilter = filter.country;
+  $: stateFilter = filter.state;
```

Comment:
Why do we need those?

Reply:
To prevent unneeded updates when `filter` is reassigned, but the values of `country` or `state` haven't actually changed

Reply:
And this prevents this? From my understanding `countryFilter` updates whenever `filter.country` updates, so `updateStates` will update whenever `filter.country` changes?

Reply:
Yes, this will prevent updating like shown in this [demo](https://svelte.dev/repl/1dde5d87b2e84090a3da4a33c126967a?version=4.2.12)

Reply:
Interesting. TIL!